p2p/nat/nat_stun.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2024 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package nat
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/pion/stun/v2"
+)
+
+// The code are from erigon p2p/nat/nat_stun.go
+// This stun server is part of the mainnet infrastructure.
+// The addr are from https://github.com/ethereum/trin/blob/master/portalnet/src/socket.rs
+const STUNDefaultServerAddr = "159.223.0.83:3478"
+
+type STUN struct {
```

Comment:
I don't think this needs to be a public type. Better make it `stun` -- we prefer to avoid adding public api surface when we can